src/nanoemoji/svg.py

```diff
@@ -340,7 +355,57 @@ def _apply_paint(
         raise NotImplementedError(type(paint))
 
 
-_XLINK_HREF_ATTR_NAME = f"{{{svg_meta.xlinkns()}}}href"
+def _attrib_apply_paint_uses(el: etree.Element) -> Set[str]:
+    # Per https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use
+    # only a few attributes of target can be overriden by <use>. However, we need
+    # only concern ourselves with the subset of those that _apply_paint might set.
+    return set(el.attrib) & _PAINT_ATTRIB_APPLY_PAINT_MAY_SET
+
+
+def _migrate_to_defs(
+    svg: SVG,
+    reused_el: etree.Element,
+    reuse_cache: ReuseCache,
+    reuse_result: ReuseResult,
+):
+    svg_defs = svg.xpath_one("//svg:defs")
+
+    if reused_el in svg_defs:
+        return  # nop
+
+    svg_use = etree.Element("use", nsmap=svg.svg_root.nsmap)
+    svg_use.attrib[_XLINK_HREF_ATTR_NAME] = f"#{reuse_result.glyph_name}"
+    # if reused_el hasn't been given a parent yet just let the <use> replace it
+    # otherwise move it from current to new parent
+    if reused_el.getparent() is None:
```

Comment:
when does reused_el have no parent? I don't think _migrate_to_defs ever meets a parent-less reused_el.
IIUC, at a first occurrence of a paint glyph, when by definition there is no (not yet) reuse, we always append the corresponding element to the parent element (svg root or g), thus by the time a reuse is found, that reused element already got a parent.
If that's the case, I think here we should assert/assume that reused_el already has a parent.

Reply:
Hm. It was definitely possible at some point but now I'm failing to come up with when that would be. I think I'm going to file a new ticket for that and leave it for this one.

Reply:
https://github.com/googlefonts/nanoemoji/issues/346